### PR TITLE
Initial pass at version 1.2.

### PR DESCRIPTION
**Pull request by waynebeaton**
Significant changes include #13 time-based progress reviews (progress
reviews are used to extend the Release Cycle); a provision #14 that all
Specification Committee Ballots are 14 days by default; and a formal
process #12 for granting exceptions.

### CODE REVIEW

File: CHANGELOG
Comment (bshannon) [inline CHANGELOG]: Lost a double quote, and need a newline at the end of the file.

File: source/process.adoc
Comment (bshannon) [inline source/process.adoc:187]: Did you decide that the TCK must be developed by an Eclipse project?
If so, should that requirement be included here?

File: source/process.adoc
Comment (bshannon) [inline source/process.adoc]: That obviously means "all decisions that are the purview of the Specification Committee", right?

File: source/process.adoc
Comment (bshannon) [inline source/process.adoc]: Are we talking about a Milestone Build of the specification document (and javadocs)?
How are those intended for use by early adopters?

Or was this intended to refer to Milestone Builds of the corresponding implementation(s)?

File: source/process.adoc
Comment (bshannon) [inline source/process.adoc]: It's the successful completion of the Release Review that ends the Release Cycle, right?
If the Release Review fails, the Release Cycle continues.

File: source/process.adoc
Comment (bshannon) [inline source/process.adoc]: Again, is this "staging of a Milestone" just talking about the specification document?
Or, like a Release Review, is it talking about all the required artifacts for a release (spec, CI, TCK)?

File: source/process.adoc
Comment (bshannon) [inline source/process.adoc]: Should there be a review process for exceptions?  If exceptions are a Big Deal, we want to allow for
the greatest transparency and inclusion, right?  And shouldn't there be a requirement to document
the exception, the voting, and the results?

**Review by bshannon — CHANGES_REQUESTED** [review]
In addition to the comments below, I'd also like a link to the Eclipse document
that describes the process for the Board of Directors to approve revisions of
this document.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: Yes. I'll change it to 

> Unless otherwise stated in this process (or a Working Group-specific derivative of this process), Super-majority approval is required for all decisions related to the development of Specifications under the purview of the Specification Committee.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: The former. Early adopters, including those who are engaged in the creation of the initial set of Compatible Implementations, will use the Milestone Builds as the basis of their work. Naturally, no compatibility claims may be made based on a Milestone Build.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: Agreed.

In the event that a Release Review fails, the project team will need to mitigate highlighted issues and re-engage.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: A Milestone Build is described as an "almost release" in the document. My intent is that the Specification Team will build what they have and distribute it via a project channel.

Care must be taken to ensure that Milestone Builds are carefully labeled as such.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: > Should there be a review process for exceptions? 
> If exceptions are a Big Deal, we want to allow for the greatest transparency and inclusion, right? 

Perhaps. But exceptions are, well, exceptional. The intent is to ensure that we don't let process get in the way of doing the right thing (i.e. following the principles that underlie this process). If somebody feels that the exception process is being used unfairly, the grievance process is described in the EDP.

A Specification Committee may choose to consult with others.

> And shouldn't there be a requirement to document the exception, the voting, and the results?

The last sentence states as much. IMHO, this includes the results of votes. We've been doing this in practice. 

Do you feel that we need to explicitly state that the voting and results must be included as part of public record? If so, please propose the specific words that you'd like to see added.

**Comment by waynebeaton** [comment]
> In addition to the comments below, I'd also like a link to the Eclipse document
> that describes the process for the Board of Directors to approve revisions of
> this document.

The build does this already. See https://www.eclipse.org/projects/efsp/#efsp-addendum-revisions

File: source/process.adoc
Comment (bshannon) [reply source/process.adoc]: Perhaps insert a few words...

... are "almost Releases" that, in combination with an implementation, are intended ...

**Comment by waynebeaton** [comment]
> Add: which is available for review by all (members/community, whomever) [under that license].

Where?

> I'd recommend we strike "intellectual property" from the purpose of extending the review period -- unless we want extensions to be strictly limited for that purpose only.

Okay.


> Distribution Form: Generally, some form of "proposed final draft," or "ballot draft" marking would be included in the ballot submission. 

Agreed. I expect that this will be done in practice. The process does make some requirements regarding the labeling of Milestone Builds (and Release Candidates by extension)

>> All communication regarding Milestone Builds must include caveats explaining that these are not official Releases. Milestone Builds and Release Candidate builds must be labeled as such (e.g. `x.yMn`, `x.yRCn`, `alpha`, `beta`, or similar).

> Must persist ... forever? 

Effectively, yes.

> What must be persisted? The contents of the submission -- the URL/location? (All of these are elements of Specifications that we have had to reconstruct, from time to time, due to various evolutionary changes.)

The Final Specification, in whatever form it takes, must persist. This is described as the combination of the specification document and related technical artifacts (e.g., API, TCK). In practice, I expect that this will manifest as a directory on a web server (see https://github.com/jakartaee/jakarta.ee/issues/302).

> Public Record is mentioned a couple of times in this document. Does this have/need a formal description? 

The specific technology will vary by Working Group and over time. I try to avoid referencing specific technologies in process documents.

> Is there a mechanism that must be used to capture this record? How does a Specification Project team know if they are recording too much, or too little in this record?

The Working Group (most likely the Specification Committee) will define this. The Specification Committee also needs to sort out (via trial and error) what sort of detail they'd like to see.

File: source/process.adoc
Comment (bshannon) [reply source/process.adoc]: I read "the exception" as "the thing we're doing differently than what the process requires".
That doesn't include the vote.  So yes, I think you need to be explicit. Add something like:

Such documentation must include the exact change to the process, the case or conditions
under which it applies, and the votes of each Specification Committee member.

This is all still ripe for abuse by a coordinated super-majority, but I guess we'll have to
worry about that if it ever happens.

**Comment by waynebeaton** [comment]
> Is language needed for the very first collection of projects which will probably be handled slightly differently since many projects proposals in EE4J were done before any of the Specification process was completed?

I'd rather not.

> Might be stronger if this were a positive statement. Maybe something like: Implementations must use approved final releases for any formal compatibility certification, formal testing, or production of test results for validation, or statements about compatibility.

Where in the document? (I feel like something funky has happened and GitHub isn't showing me where in the document you're commenting).

> Does the process distinguish what a "Release Candidate" must contain? Or is this just left to the project team to decide what distinguishes milestone releases from one another?

No. Only that later Milestone Builds may be referred to as "Release Candidates". In practice, I expect that "Release Candidates" are feature complete (on parity with a Final Specification).

> Is the intent here that they are staged for feedback, or is it important to have require some form of distribution limitation?

Yes. The section titled "Milestone Builds" attempts to cover this.

> As separate tasks, Release Review and Specification Version Ratification -- I read this as allowing concurrency. Since both are "pass/fail" operations, the Specification Version will fail to be ratified, if either of these fails. Do either of these processes feed into the other (I don't think so, but I want to check)?

A positive result from a Specification Committee ballot is required to approve a review. In practice, for Progress and Release Reviews, the EMO will need IP Log approval, PMC approval, and Specification Committee approval. There are no timing restrictions on any of these elements other than that they must be completed in order to approve a review.

A Specification Version is ratified (and becomes a "Final Specification") upon the successful completion of a Release Review.

> I would suggest: The Specification Committee Project Proposal Ballot and the Creation Review [may] run in parallel.

Maybe. The real relationship, however, is that the a positive result from the Specification Committee Ballot is required to approve the Creation Review. I'll have to think about this a bit...

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: The teams building Compatible Implementations are the early adopters. 

I suppose that organizations building servlets and things might also be adopters, but that's not what I intend.

**Comment by bshannon** [comment]
> > In addition to the comments below, I'd also like a link to the Eclipse document
> > that describes the process for the Board of Directors to approve revisions of
> > this document.
> 
> The build does this already. See https://www.eclipse.org/projects/efsp/#efsp-addendum-revisions

Hmmm...  I was hoping the statement at the top would be able to link to this.

And looking at the revision process, did we skip step 1, or did I miss it?
It seems we're already deep into "authors an updated document".

File: source/process.adoc
Comment (bshannon) [reply source/process.adoc]: Ya, I think of the people *using* the non-final implementations as the early adopters.
The people building the implementations are just the implementors.  I don't classify
them as "early" or "late".

I guess it's all relative, but the important adopters are the people using the implementations.
If the people building the implementations are adopters of the spec, then what do we call
the people using the implementations?

Comment (waynebeaton) [reply source/process.adoc]: Based on Ed's comments, I'm going to revert this change (i.e. "should" will be turned back to "must").

Coming up with the right words here is hard. I think that "implementors" is probably the best. How do we express that?

"... intended to guide and inform implementors, validate build and deployment practices, and solicit feedback." ?

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: > Such documentation must include the exact change to the process, the case or conditions under which it applies, and the votes of each Specification Committee member.

I'll add that sentence.

> This is all still ripe for abuse by a coordinated super-majority, but I guess we'll have to worry about that if it ever happens.

We ultimately have to answer to the community. We have a responsibility to operate in good faith.

File: CHANGELOG
Comment (kwsutter) [inline CHANGELOG]: Per the discussion on today's Spec Committee call, I'd like to understand why we're making this requirement more restrictive.  We're trying to entice more projects (ie. MicroProfile) to adopt the EFSP, but if we're making it more restrictive with longer required ballot periods, then that's not going to help.  The original EFSP indicated a minimum review period of one week.  We can clarify that to 7 days.  That would be my preference.

File: source/process.adoc
Comment (kwsutter) [inline source/process.adoc]: Maybe this "DRAFT" can be changed into a doc variable so that it can be modified as part of the final build process for the EFSP?  We do that with our MicroProfile spec documents.  During our build process, we parameterize the [Draft/Final] version of the release and that is used in our pdf and html generation of the Spec document.  Something similar for the EFSP might be nice.

File: source/process.adoc
Comment (kwsutter) [inline source/process.adoc:187]: Should we qualify this with "under an approved Open Source License"?  Or, since it's capitalized, is it assumed that the definitions in the EFSP take precedence?

File: source/process.adoc
Comment (kwsutter) [inline source/process.adoc]: I already posted my views in the ChangeLog section and in the recent email exchange.  We should not make the EFSP more restrictive.  Leave that up to the defined derivatives.

**Review by kwsutter — CHANGES_REQUESTED** [review]
In my mind, the only required change was the change of the required ballot period from one week to 14 days.  I'd like that changed back to 7 days as the default and minimum for the EFSP.

**Comment by bshannon** [comment]
Another issue came to mind during today's meeting...

We need to make it clear that the exception process may not be used to override a member's
request to extend the ballot period.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc:187]: I'd rather defer this for now.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: Per our discussion during the Specification Committee call today, I think that this may end up being far too broad. At this point, the process is no worse without it, so I'm going to remove it.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: Good idea. Can I assume that you've set up those builds so that they say "DRAFT" unless specified otherwise?

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc:187]: The capitalization indicates that the defined term is intended.

**Comment by waynebeaton** [comment]
> We need to make it clear that the exception process may not be used to override a member's
> request to extend the ballot period.

Added:

"This exception process may not be used to override a Specification Committee member's request to extend the ballot period."

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc:187]: I've opened #16 to track this.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: I've opened #17. Defer for now.

File: CHANGELOG
Comment (waynebeaton) [reply CHANGELOG]: I've changed this to a seven day default.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: I've changed it to a simple seven day default.

File: source/process.adoc
Comment (waynebeaton) [reply source/process.adoc]: Moved to issue #18.

**Review by kwsutter — APPROVED** [review]
Thanks, Wayne!

**Review by ivargrimstad — APPROVED** [review]

**Pull request merged**
